walletrpc/service.proto

```diff
@@ -26,10 +26,10 @@ message BlockID {
 // BlockRange specifies a series of blocks from start to end inclusive.
 // Both BlockIDs must be heights; specification by hash is not yet supported.
 //
-// A server may optionally prune `CompactBlocks` returned to include only data
-// relevant to the requested pool types. If no pool types are specified, the
-// server should default to the legacy behavior of returning only data relevant
-// to the shielded (Sapling and Orchard) pools.
+// The server must prune `CompactBlocks` returned to include only data relevant
+// to the requested pool types. If no pool types are specified, the server
+// should default to the legacy behavior of returning only data relevant to the
+// shielded (Sapling and Orchard) pools.
 message BlockRange {
```

Comment:
Why are these semantics being changed? What has changed since the discussions in #1?

Reply:
My main concern now is that I expect "The server must" language to be normative on how clients validate server responses, but in practice I expect clients will not validate in this particular instance for my above-mentioned reasons of practicality. That to me implies this is the wrong language to use in the specification; instead, we should clarify how both servers and clients should behave around this new option.